### PR TITLE
[RDBMS] Update list-skus, create command location setting and replica command

### PR DESCRIPTION
**Pull request by DaeunYim**
**Description**

- Change list-skus logic for single availability zone checking
- Added replica validation
- Fixed bug of create command location setting
- private access server restore command network resources to be same as source server's

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py [hunk 1]
```diff
@@ -365,11 +365,11 @@ def _flexible_server_params(command_group):
         )
 
         mysql_high_availability_arg_type = CLIArgumentType(
-            arg_type=get_enum_type(['ZoneRedundant', 'SameZone', 'Disabled']),
+            arg_type=get_enum_type(['ZoneRedundant', 'SameZone', 'Disabled', 'Enabled']),
             options_list=['--high-availability'],
             help='Enable (ZoneRedundant or SameZone) or disable high availability feature. '
                  'Default value is Disabled. High availability can only be set during flexible server create time. '
-                 'Allowed values: ZoneRedundant, SameZone, Disabled'
+                 'Allowed values: ZoneRedundant, SameZone, Disabled, Enabled'
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:372]: CLI core will generate `Allowed values` automatically for enum type arguments, you can omit this line.

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py [hunk 2]
```diff
@@ -50,7 +50,10 @@ def flexible_server_create(cmd, client,
         cf_availability=cf_mysql_check_resource_availability, cf_private_dns_zone_suffix=cf_mysql_flexible_private_dns_zone_suffix_operations, logging_name='MySQL', command_group='mysql', server_client=client,
         location=location)
 
+    # Process parameters
     server_name = server_name.lower()
+    if high_availability and high_availability.lower() == 'enabled':
+        high_availability = 'ZoneRedundant'
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py:56]: Why do you need to add `Enabled` if it's only an alternative of `ZoneRedundant`?

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py (hunk 1 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_params.py]: Aha. thanks!

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py (hunk 2 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py:56]: PMs asked me to make it downward compatible.. so adding back the 'enabled' parameter. What do you think is the best way to deprecate acceptable parameter?

Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py:56]: Got it. CLI has deprecating mechanism for some argument but no for argument acceptable enums. I guess you can add a warning here to let customer know that `Enabled` is going to be deprecated, use `ZoneRedundant` instead.

Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py:56]: I see. Let me add that! Thank you :)

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**